bigchaindb/tep/upsert_validator.py

```diff
@@ -0,0 +1,216 @@
+        return self._is_same_topology(current_validators, tx_election.outputs)
+
+    def vote(self, tx_election_id, node_key_path):
+        """Vote on the given `election_id`"""
+
+        tx_election = self.bigchain.get_transaction(tx_election_id)
+        node_key = load_node_key(node_key_path)
+        (tx_vote_input,
+         voting_power) = self._get_vote(tx_election, node_key.public_key)
+
+        election_id = self.election_id(tx_election_id)
+
+        # NOTE: Node will spend all its allocated votes
+        tx_vote = Transaction.transfer(tx_vote_input,
+                                       [([election_id], voting_power)],
+                                       metadata={'type': 'vote'},
+                                       asset_id=tx_election.id)\
+                             .sign([node_key.private_key])
+        return self._execute_action(tx_vote)
+
+    def is_valid_vote(self, tx_vote):
+        """Validate casted vote.
+
+        NOTE:
+        * We don't need to check the amount of votes since the same has been
+          validated when the election was initiated and a voter can't spend
+          more votes than allocated to them in the election proposal.
+
+        * A voter can choose to spend their votes whichever way they like i.e.
+          they can tranfer their votes to `election_id` or someone else. The only
+          thing that counts is whether the number of votes casted to the `election_id`
+          reach a majority of >2/3. It is the responsibility of the voter to
+          spend their votes wisely.
+
+        * The byzantine behavour cannot be amplified by sending votes to non-validators
+          i.e. if 1/3 of the network is byzantine then their no way to amplify this behavour
+          by merely transfering votes to other byzantine nodes.
+        """
+
+        return True
+
+    def is_concluded(self, tx_election_id, current_votes=[]):
+        """Check if the Election with `election_id` has gained majority vote
+
+        NOTE:
+        * An election is concluded only if the >2/3 of network has casted their vote.
+
+        * If the network topology has changed from the time the election was
+          proposed to the time when the majority was achieved then the validator
+          updates are not applied. The election can only be concluded if the
+          network topology reverts back to the time when the election was proposed.
+
+        * If an election cannot be concluded because of topology changes then a new
+          election can be initiated.
+        """
+
+        tx_election = self.bigchain.get_transaction(tx_election_id)
+        current_validators = self._validators()
+
+        # NOTE: Check if network topology is exactly same to when the election was initiated
+        if not self._is_same_topology(current_validators, tx_election.outputs):
+            return (False, 'inconsistant_topology')
```

Comment:
`is_concluded` suggests the function returns a boolean. I would either keep it simple and not return any statuses or rename it into something like `get_election_status`, does it make sense?

There is a typo - inconsist**e**nt_topology.